src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -586,9 +594,19 @@ private void GenerateW3CId()
                 if (Parent != null && Parent.IdFormat == ActivityIdFormat.W3C)
                     _traceId = Parent.TraceId;
                 else if (_parentId != null && IsW3CId(_parentId))
-                    _traceId = new ActivityTraceId(_parentId.AsSpan(3, 32));
-                else
+                {
+                    try
+                    {
+                        _traceId = new ActivityTraceId(_parentId.AsSpan(3, 32));
+                    }
+                    catch
```

Comment:
Why is this catch needed? What here can fail?

Reply:
ActivityTraceId ctor throws if _parentId is invalid

Reply:
The ActivityTraceId constructor that takes a span<char> can fail if the format is wrong (it is not exactly 32 bytes of lower case hex).     The problem is that IsW3CId method was originally intended to make a choice based on the ID string what format it was.   It did not have to be 'perfect', just enough to make a format choice (we just look that it is the right length and starts with a digit).    We want that check to be cheap.   

Here we are using it for complete validation.  The try-catch is OK (but I must admit having a 'tryParseTraceId' that wraps it feels a bit better.   Still I am OK with the inline try-catch.  

We do something similar with IsW3Cid in ParentSpanId.    We should put at catch around that as well.  

The other uses of the ActivityTraceId and ActivitySpanId constructors seems to be in places where it really is a programmer error if they fail (in fact I think they may be unreachable), so we can leave those alone.  

@lmolkova: can you add a try-catch around the ActivitySpanId call in ParentSpanId (it should return the empty ID if the parentID is a bad format.

Reply:
@vancem done

Reply:
> Here we are using it for complete validation. The try-catch is OK (but I must admit having a 'tryParseTraceId' that wraps it feels a bit better. Still I am OK with the inline try-catch.

It'll be very rare then that this throws?  If so, ok.  If it might be at all common, though, it'd be better for it to be a TryParse, and it looks like that'd be trivial to do.

Reply:
it should be very rare  - only when someone violates standard

Reply:
Ok, thanks.